pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/service_proxy_transport.go

```diff
@@ -0,0 +1,203 @@
+	newRestConfig := &restclient.Config{
+		TLSClientConfig: restclient.TLSClientConfig{
+			ServerName: r.serverName,
+			CAData:     r.caBundle,
+		},
+	}
+
+	rt, err := restclient.TransportFor(newRestConfig)
+	if err != nil {
+		return nil, err
+	}
+	return rt.RoundTrip(req)
+}
+
+// These must align with the service-ca operator configuration
+const (
+	caBundleDataKey              = "cabundle.crt"
+	serviceCABundleNamespace     = "openshift-service-cert-signer"
+	serviceCABundleConfigMapName = "signing-cabundle"
+)
+
+// serviceCABundleUpdater runs a simple controller to keep rt.caBundle updated with CAs from the service-ca controller.
+type serviceCABundleUpdater struct {
+	// Initial CA bundle that CA updates are tacked on to.
+	startingHandlerCA []byte
+	// RoundTripper that utilizes the updated CA bundle.
+	rt *serviceCABundleRoundTripper
+
+	lister      listersv1.ConfigMapLister
+	queue       workqueue.RateLimitingInterface
+	hasSynced   cache.InformerSynced
+	syncHandler func(serviceKey string) error
+}
+
+func isServiceCABundleConfigMap(configMap *v1.ConfigMap) bool {
+	return configMap.Namespace == serviceCABundleNamespace && configMap.Name == serviceCABundleConfigMapName
+}
+
+// addCABundle is the informer's AddFunc.
+func (u *serviceCABundleUpdater) addCABundle(obj interface{}) {
+	cm := obj.(*v1.ConfigMap)
+	if !isServiceCABundleConfigMap(cm) {
+		return
+	}
+
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(cm)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", cm, err)
+		return
+	}
+
+	glog.V(4).Infof("serviceCABundleUpdater controller: queuing an add of %v", key)
+	u.queue.Add(key)
+}
+
+// updateCABundle is the informer's UpdateFunc.
+func (u *serviceCABundleUpdater) updateCABundle(old, cur interface{}) {
+	cm := cur.(*v1.ConfigMap)
+	if !isServiceCABundleConfigMap(cm) {
+		return
+	}
+
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(cm)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", cm, err)
+		return
+	}
+
+	glog.V(4).Infof("serviceCABundleUpdater controller: queuing an update of %v", key)
+	u.queue.Add(key)
+}
+
+// processNextWorkItem processes the queued items.
+func (u *serviceCABundleUpdater) processNextWorkItem() bool {
+	key, quit := u.queue.Get()
+	if quit {
+		return false
+	}
+	defer u.queue.Done(key)
+
+	err := u.syncHandler(key.(string))
+	if err == nil {
+		u.queue.Forget(key)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %v", key, err))
+	u.queue.AddRateLimited(key)
+
+	return true
+}
+
+// Run runs the controller until stopCh is closed.
+func (u *serviceCABundleUpdater) Run(stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer u.queue.ShutDown()
+	glog.V(2).Infof("starting serviceCABundleUpdater controller")
+
+	if !cache.WaitForCacheSync(stopCh, u.hasSynced) {
+		return
+	}
+
+	go wait.Until(u.runWorker, time.Second, stopCh)
+	<-stopCh
+	glog.V(2).Infof("stopping serviceCABundleUpdater controller")
```

Comment:
Sigh, is there something else we can do that does not involve embedding a controller :(

Reply:
> Sigh, is there something else we can do that does not involve embedding a controller :(

Yes.  The lister could be called and if the resourceVersion has changed since it was last observed, the transport could be rebuilt. 

I don't really care which way this does, but I  think that would work.